### PR TITLE
fix: guard ServicenowError.Error against nil main error and detail

### LINKED ISSUE #514: `core.ServiceNowError.Error()` panics with nil-pointer dereference when API error response omits message/detail
### Summary

`ServiceNowError.Error()` in `core/service_now_error.go` can panic instead of returning an error string, when a ServiceNow API error response doesn't populate the fields it expects. This turns a normal HTTP-error response into an unrecoverable panic for the caller.

### Reproduction

```go
client.Now().Table("incident").Get(ctx, nil)
```

against (or mocked to return) a ServiceNow error response whose `error` object has neither `message` nor `detail` populated, e.g.:

```json
{
  "error": {},
  "status": "failure"
}
```

### Expected behavior

`Error()` returns some reasonable string (e.g. a fallback message) even when the response body is missing fields — callers should never see a raw panic surfacing from error-formatting code.

### Actual behavior

`Error()` panics with a nil-pointer dereference. There are actually two separate panic sites here:

1. `core/service_now_error.go:139-140`:
   ```go
   details, _ := mainErr.GetDetail()
   return *details
   ```
   If `msg` is nil and `GetDetail()` also returns a nil `*string` (both `message` and `detail` absent/unset in the response), this dereferences a nil pointer directly.

2. `core/service_now_error.go:134-135`:
   ```go
   mainErr, _ := exc.GetError()
   msg, _ := mainErr.GetMessage()
   ```
   `exc.GetError()` returns `(MainErrorable, error)` — if the response body has no `error` key at all (or deserialization otherwise leaves it unset), `mainErr` is a nil interface, and calling `mainErr.GetMessage()` panics immediately, before even reaching the `details` line.

Both stem from the same root cause: `Error()` ignores every error return value from `GetError()`/`GetMessage()`/`GetDetail()` and assumes success always yields non-nil pointers/interfaces.

### Root cause

```go
func (exc *ServiceNowError) Error() string {
	mainErr, _ := exc.GetError()
	msg, _ := mainErr.GetMessage()
	if msg != nil {
		return *msg
	}
	details, _ := mainErr.GetDetail()
	return *details
}
```

No nil-guard on `mainErr` before calling methods on it, and no nil-guard on `details` before dereferencing.

### Suggested fix

- Nil-check `mainErr` (via `internal/conversion.IsNil`, consistent with the nil-guard convention used elsewhere in this package) before calling `GetMessage()`/`GetDetail()` on it.
- Nil-check `details` before dereferencing.
- Fall back to a sensible default string (e.g. "unknown service-now error" or similar) when none of message/detail/mainErr are populated, rather than an empty string, so `Error()` always returns something useful.
- Add a unit test covering: (a) both message and detail nil, (b) `error` key entirely absent from the response.

### PR DESCRIPTION
**Pull request by SAY-5**
## Summary
- `ServicenowError.Error()` panicked with a nil-pointer dereference when a ServiceNow error response omitted the `error` object (nil `mainErr`) or had neither `message` nor `detail` set (`*details` deref on a nil pointer).
- Guards `mainErr` with the package `IsNil` helper and checks `details` before dereferencing, falling back to a fixed string so `Error()` never panics.

Fixes #514

## Testing
- `go test ./internal/new/`

### CODE REVIEW

File: internal/new/service_now_error.go [hunk 1]
```diff
@@ -68,10 +68,16 @@ func (exc *ServicenowError) setError(mainError MainErrorable) error {
 
 func (exc *ServicenowError) Error() string {
```
Comment (michaeldcanady) [inline internal/new/service_now_error.go:69]: Hey, thank you again for your first PR (and wow, you were quick with this one) nice work!

One small note: this is one of the rare cases where I’d go against the usual early‑return pattern. Instead of:

```golang
mainErr, _ := exc.GetError()
if IsNil(mainErr) {
    return "unknown service-now error"
}
...
return "unknown service-now error"
```

I’d flip the condition:

```golang
mainErr, _ := exc.GetError()
if !IsNil(mainErr) {
    ...
}
return "unknown service-now error"
```

This avoids needing multiple nested checks and prevents duplicating the `"unknown service-now error"` string. Very minor, but it keeps the flow a bit cleaner.

**Review by michaeldcanady — APPROVED** [review]
